src/runtime/monitor/src/mon/compartment.rs

```diff
@@ -65,4 +70,90 @@ impl CompartmentMgr {
         // Unwrap-Ok: this instance is always present.
         self.get_mut(MONITOR_INSTANCE_ID).unwrap()
     }
+
+    /// Get an iterator over all compartments.
+    pub fn compartments(&self) -> impl Iterator<Item = &RunComp> {
+        self.instances.values()
+    }
+
+    /// Get an iterator over all compartments (mutable).
+    pub fn compartments_mut(&mut self) -> impl Iterator<Item = &mut RunComp> {
+        self.instances.values_mut()
+    }
+}
+
+impl super::Monitor {
+    /// Get CompartmentInfo for this caller. Note that this will write to the compartment-thread's
```

Comment:
Why does this, and similar read-only operations, need to write into the compartment buffer?